source/ghost-text.js

```diff
@@ -253,19 +253,39 @@ async function updateCount() {
 	}
 }
 
+const registeredFrames = new Set([document]);
+function injectCSS(root) {
+	// Injects ghost-text.css into iframe document roots
+	if (!registeredFrames.has(root)) {
+		const cssLink = root.createElement("link")
+		cssLink.href = chrome.runtime.getURL('ghost-text.css');
+		cssLink .rel = "stylesheet";
+		cssLink.type = "text/css";
+		registeredFrames.add(root);
+		root.body.appendChild(cssLink);
+	}
+}
+
 const selector = `
 	textarea,
 	[contenteditable=""],
 	[contenteditable="true"]
 `;
-function registerElements() {
-	for (const element of document.querySelectorAll(selector)) {
+
+
+function registerElements(root = document) {
+	injectCSS(root); // only if the CSS hasn't already been injected
+	for (const element of root.querySelectorAll(selector)) {
 		// TODO: Only handle areas that are visible
 		//  && element.getBoundingClientRect().width > 20
 		if (!knownElements.has(element)) {
 			knownElements.set(element, new GhostTextField(element));
 		}
 	}
+	for (const iframe of root.getElementsByTagName("iframe")) {
```

Comment:
You can move the CSS injection here instead of tracking it manually via `registeredFrames`

```suggestion
	for (const iframe of root.getElementsByTagName("iframe")) {
		injectCSS(iframe)
```

Reply:
I still need registeredFrames because an iframe might be observed multiple times. Renamed to injectCSSOnce for clarity.